subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -442,73 +442,69 @@ static void wifi_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+	static struct option long_options[] = {{"ssid", required_argument, 0, 's'},
+					       {"passphrase", required_argument, 0, 'p'},
+					       {"key-mgmt", required_argument, 0, 'k'},
+					       {"ieee-80211w", required_argument, 0, 'w'},
+					       {"bssid", required_argument, 0, 'm'},
+					       {"band", required_argument, 0, 'b'},
+					       {"channel", required_argument, 0, 'c'},
+					       {"help", no_argument, 0, 'h'},
+					       {0, 0, 0, 0}};
+	int opt_index = 0;
+	uint8_t band;
+	const uint8_t all_bands[] = {
+		WIFI_FREQ_BAND_2_4_GHZ,
+		WIFI_FREQ_BAND_5_GHZ,
+		WIFI_FREQ_BAND_6_GHZ
+	};
+	bool found = false;
+	char bands_str[MAX_BANDS_STR_LEN] = {0};
+	size_t offset = 0;
 
 	/* Defaults */
 	params->band = WIFI_FREQ_BAND_UNKNOWN;
 	params->channel = WIFI_CHANNEL_ANY;
 	params->security = WIFI_SECURITY_TYPE_NONE;
+	params->mfp = WIFI_MFP_OPTIONAL;
 
-	/* SSID */
-	params->ssid = argv[0];
-	params->ssid_length = strlen(params->ssid);
-	if (params->ssid_length > WIFI_SSID_MAX_LEN) {
-		PR_WARNING("SSID too long (max %d characters)\n",
-			   WIFI_SSID_MAX_LEN);
-		return -EINVAL;
-	}
-
-	/* Channel (optional: STA, mandatory: AP) */
-	if ((idx < argc) && (strlen(argv[idx]) <= 3)) {
-		uint8_t band;
-		long channel = strtol(argv[idx], &endptr, 10);
-		const uint8_t all_bands[] = {WIFI_FREQ_BAND_2_4_GHZ,
-					WIFI_FREQ_BAND_5_GHZ,
-					WIFI_FREQ_BAND_6_GHZ};
-		bool found = false;
-		char bands_str[MAX_BANDS_STR_LEN] = {0};
-		size_t offset = 0;
-
-		if (*endptr != '\0') {
-			PR_ERROR("Failed to parse channel: %s: endp: %s, err: %s\n",
-				 argv[idx],
-				 endptr,
-				 strerror(errno));
-			return -EINVAL;
-		}
-
-		if (iface_mode == WIFI_MODE_INFRA) {
-			if (channel < 0) {
-				/* Negative channel means band */
-				switch (-channel) {
-				case 2:
-					params->band = WIFI_FREQ_BAND_2_4_GHZ;
-					break;
-				case 5:
-					params->band = WIFI_FREQ_BAND_5_GHZ;
-					break;
-				case 6:
-					params->band = WIFI_FREQ_BAND_6_GHZ;
-					break;
-				default:
-					PR_ERROR("Invalid band: %ld\n", channel);
-					return -EINVAL;
-				}
-			}
-		} else {
-			if (channel < 0) {
-				PR_ERROR("Invalid channel: %ld\n", channel);
+	while ((opt = getopt_long(argc, argv, "s:p:k:w:b:c:m:h", long_options, &opt_index)) != -1) {
+		state = getopt_state_get();
+		switch (opt) {
+		case 's':
+			params->ssid = optarg;
+			params->ssid_length = strlen(params->ssid);
+			if (params->ssid_length > WIFI_SSID_MAX_LEN) {
+				PR_WARNING("SSID too long (max %d characters)\n",
+					    WIFI_SSID_MAX_LEN);
 				return -EINVAL;
 			}
-		}
-
-		if (channel > 0) {
+			break;
+		case 'k':
+			params->security = atoi(optarg);
+			if (params->security) {
+				secure_connection = true;
+			}
+			break;
+		case 'p':
+			if (secure_connection) {
+				params->psk = optarg;
+				params->psk_length = strlen(params->psk);
+			} else {
+				PR_WARNING("Passphrase provided without security configuration\n");
+			}
+			break;
+		case 'c':
```

Comment:
/subsys/net/l2/wifi/wifi_shell.c:507:4: error: a label can only be part of a statement and a declaration is not a statement
  507 |    long channel = strtol(optarg, &endptr, 10);